test/suite.js

```diff
@@ -72,6 +72,19 @@ class Suite {
     const test = new Test(name, cb)
     this._queue.push(test)
   }
+
+  testAsync (name, action) {
```

Comment:
i apologize the tests here aren't written w/ a modern node testing framework.  There's this cost-benefit thing I keep running into where it's like "Should I re-write all these tests into jest/mocha/etc? They already work and I *know* they work...but maybe rewriting will introduce some breakage."  The original tests in the repo were written before mocha or jest existed so it's kinda like...well....if it ain't broke...otoh I feel like it's probably more confusing to contribute to.  I also at the time, coming from ruby, thought i'd be 'clever' and attach a bunch of stuff to the global namespace like assert and other things.  That was a mistake. :(

Reply:
No worries. I've thought about cleaning up some of them more than once and it's always the same conclusion. Maybe having a separate harness for things going forward as it's definitely easier to write async tests with an async harness. Doesn't even have to be be full on mocha/jest/whatever ... just having something to call out to with async actions should be enough.

I like how you can easily run any one of the tests in this project via "node path/to/test.js".